src/EPR.Calculator.API/Controllers/CalculatorController.cs

```diff
@@ -52,5 +55,33 @@ public IActionResult GetCalculatorRun()
             // TODO: Return the details of a particular run
             return new OkResult();
         }
+
+        [HttpGet]
+        [Route("calculatorRunByName/{name}")]
+        public IActionResult GetCalculatorRunByName([FromRoute] string name)
+        {
+            if (!ModelState.IsValid)
+            {
+                return StatusCode(StatusCodes.Status400BadRequest, ModelState.Values.SelectMany(x => x.Errors));
+            }
+
+            var calculatorRun = _context.CalculatorRuns.Where(run => run.Name.ToLower() == name.ToLower());
+
+            if (calculatorRun.IsNullOrEmpty())
+            {
+                return new ObjectResult("No data found for this calculator name") { StatusCode = StatusCodes.Status404NotFound };
+            }
+
+            try
+            {
+                var currentRunDetails = CalculatorRunMapper.Map(calculatorRun.First());
+
+                return new ObjectResult(currentRunDetails) { StatusCode = StatusCodes.Status200OK };
```

Comment:
We have agreed that we will not send any response. Just 200OK
We do not need to map.

Reply:
Thanks a lot Uday,
Agreed but in one of the AC1 it mentioned that need details for front end . I had chat with John and Trilok this was the information required. I hope that make sense.

Reply:
Front end will not be using the response data. All it cares if it get 200 or 404. @johnbenjamin-defra  please confirm. Thanks

Reply:
@udaydefra - when discussed with @rekhadevieviden we went with what was there in the AC and expected that the details could be useful for the future.
It could be that the AC was not updated during the refinement.
If we are not expecting a response data then we have to modify the AC and do the code changes accordingly.